### PR TITLE
Upgrade to Java 21, fix Docker image, add server integration test

### PR DESCRIPTION
**Pull request by StenAL**
This PR is a collection of smaller changes:
- The project is upgraded to the latest Java LTS version, Java 21. Features added since Java 17 include things like pattern matching, unnamed variables, and UTF-8 as the charset.
- An integration test with a mocked client was added to the server to verify it's working as expected. This will be useful later when upgrading the server to Netty 4.1 to avoid breaking anything.
  - The testing revealed that previously messages sent from the server non-deterministic, i.e. sometimes TrackSets and Tracks were sent in one order and sometimes in another. Now these are sorted in the server to ensure the same message is sent every time and tests pass. This also fixes a bug where championship games did not run in the intended order and the games could start with the hardest tracks in the championship first.
- The server Docker image was fixed and is now built in CI to ensure it won't break again. I accidentally broke it when upgrading the project to Java 17 earlier.
- A bunch of dependencies were upgraded and warnings fixed to make build logs more readable.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,3 +32,7 @@ jobs:
         run: sudo apt-get install -y genisoimage
       - name: Build with Maven
         run: mvn -B install --file pom.xml
+      - name: Build Docker image
+        uses: docker/build-push-action@v5
+        with:
+          tags: philippvk/minigolf:latest
```
Comment (PhilippvK) [inline .github/workflows/build.yml:38]: I guess I will have to add the access token as a repo secret?

Comment (PhilippvK) [reply .github/workflows/build.yml:38]: Is this pushing to docker.io or https://quay.io/repository/philippvk/minigolf?

Comment (StenAL) [reply .github/workflows/build.yml:38]: Only if you want CI to push the image to a Docker repo. In it's current form, CI will just build the image to make sure it doesn't break unexpectedly.

**Comment by PhilippvK** [comment]
@StenAL sorry for the delay, I will try to test your changes tomorrow and provide a review!

**Comment by PhilippvK** [comment]
Looks good, thank you for the patches!

**Pull request merged**